aspnetcore/host-and-deploy/scaling-aspnet-apps/scaling-aspnet-apps.md

```diff
@@ -0,0 +1,352 @@
+ms.author: alexwolf
+ms.custom: mvc
+ms.date: 08/25/2022
+---
+# Deploying and scaling an ASP.NET Core app on Azure Container Apps
+
+Web applications deployed to Azure should be dynamically scalable in order to leverage various benefits of the cloud, such as elasticity. ASP.NET Core apps meet these requirements, but developers must consider certain architectural patterns and configurations to ensure success. This tutorial demonstrates how to deploy a scalable Razor Pages app to Azure Container Apps by completing the following tasks:
+
+1. [Setup the sample project](#1-setup-the-sample-project)
+1. [Deploy the app to Azure Container Apps](#2-deploy-the-app-to-azure-container-apps)
+1. [Scale and troubleshoot the app](#3-scale-and-troubleshoot-the-app)
+1. [Create the Azure Services](#4-create-the-azure-services)
+1. [Connect the Azure services](#5-connect-the-azure-services)
+1. [Configure and redeploy the app](#6-configure-and-redeploy-the-app)
+
+In some cases, simple ASP.NET Core apps are able to scale without special considerations. However, some apps require additional configurations to utilize certain framework features or architectural patterns, including the following:
+
+* **Secure form submissions**: Razor Pages, MVC and Web API apps often rely on form submissions. By default these apps use cross site forgery tokens and framework data protection services to secure requests. When deployed to the cloud, these apps must be configured to use a managed data protection service in a secure, centralized location.
+
+* **SignalR circuits**: Blazor Server applications require the use of a centralized Azure SignalR service in order to scale properly and securely. These services also utilize the data protection services mentioned previously.
+
+* **Centralized caching or state management services**: Scalable applications may use Azure Cache for Redis to provide distributed caching. Azure storage or database services may be needed to manage state for frameworks such as Orleans, which can assist in writing stateful apps that scale across many different app instances.
+
+> [!NOTE]
+> The steps ahead demonstrate how to properly address these concerns by deploying a scalable app to Azure Container Apps. Most of the concepts in this tutorial also apply when scaling Azure App Service instances.
+
+## 1) Setup the sample project
+
+You can use the GitHub Explorer application to follow along with this tutorial. Clone the application from GitHub using the following command:
+
+```dotnetcli
+git clone "https://github.com/MicrosoftDocs/mslearn-dotnet-debug-visual-studio-app-service.git"
+```
+
+The sample app uses a search form to browse GitHub repositories by name. The form relies on the built-in ASP.NET Core data protection services to handle anti-forgery concerns. By default, when the app scales horizontally on Container Apps, the data protection service will throw an exception. You'll explore and solve this issue in the steps ahead.
```

Comment:
Check out the MS style guide, FT is not allowed without an exception.

Reply:
I'm probably missing something obvious here, but I'm not following - what is FT?

Reply:
Sorry, Future Tense, look for `will`

Reply:
the data protection service ~will~ throws an exception

MT (machine translates) much better, is more clear, less words

Reply:
Done